### DISCUSSION TITLE
How to calculate future SPI using historical and projected precipitation data

### DISCUSSION THREAD

**Discussion by DnyaneshWarade** [monocongo/climate_indices#553] [Q&A]
Future SPI calculation entails determining the deviation of projected precipitation values  from the reference historical period (mean and standard deviation of historical period to be considered).

My query is can the current implementation of spi in the climate_indices package calculate future spi as per the above description?

For example, if my reference period is 1980-2023 and I have monthly precipitation values for this period and future period is 2025-2050 and I have projected rainfall values for this period, can I use the below to calculate future spi values? If yes, will providing 1980-2050 precipitation values in np.ndarray, 1980 as data_start_year, 1980 as calibration_year_initial, 2023 as calibration_year_final, work?

def spi(
    values: np.ndarray,
    scale: int,
    distribution: Distribution,
    data_start_year: int,
    calibration_year_initial: int,
    calibration_year_final: int,
    periodicity: compute.Periodicity,
    fitting_params: Dict = None,
) -> np.ndarray:

**Comment by monocongo** [comment]
Yes, this seems reasonable. Please give it a try and report back on how it worked out for you.
